docs/run_nydus_in_kubernetes.md

```diff
@@ -0,0 +1,128 @@
+# Run Nydus snapshotter in Kubernetes
+
+This document will introduce how to run Nydus snapshotter in Kubernetes cluster, you can use helm to deploy Nydus snapshotter container.
+
+**NOTE:** This document is mainly to allow everyone to quickly deploy and experience Nydus snapshotter in the Kubernetes cluster. You cannot use it as a deployment and configuration solution for the production environment.
+
+## Setup Kubernetes using kind
+
+[kind](https://kind.sigs.k8s.io/) is a tool for running local Kubernetes clusters using Docker container “nodes”.
+kind was primarily designed for testing Kubernetes itself, but may be used for local development or CI.
+
+First, we need to prepare a configuration file(`kind-config.yaml`) for kind to specify the devices and files we need to mount to the kind node.
+
+```yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+  ipFamily: dual
+nodes:
+  - role: control-plane
+    image: kindest/node:v1.23.4
+    extraMounts:
+      - hostPath: ./containerd-config.toml
+        containerPath: /etc/containerd/config.toml
+      - hostPath: /dev/fuse
+        containerPath: /dev/fuse
+```
+
+Next, we also need a config for containerd(`containerd-config.toml`).
+
+```toml
+version = 2
+[debug]
+  level = "debug"
+
+[plugins."io.containerd.grpc.v1.cri".containerd]
+  discard_unpacked_layers = false
+  disable_snapshot_annotations = false
+  snapshotter = "overlayfs"
```

Comment:
Should we change the CRI snapshotter to `nydus`?

Reply:
Can we do that? First, we should pull the nydus-snapshotter image and start the container, then the nydus snapshotter can be used.

Reply:
I see. Nydus-snapshotter is deployed as DaemonSet and packed in an OCI container image.
Is it the same containerd pulling  OCI image packing nydus-snapshotter and nydus image?
If yes, I don't understand how the containerd can choose nydus as snapshotter when pulling the nydus image `ghcr.io/dragonflyoss/image-service/nginx:nydus-latest` since containerd can only configure statically the snapshotter type here

Reply:
In the current helm chart, the snapshotter will be pulled and unpacked by overlayfs, and then change the global snapshotter to nydus and restart the containerd after the snapshotter started.